packages/grafana-ui/src/components/Gauge/Gauge.tsx

```diff
@@ -112,7 +103,17 @@ export class Gauge extends PureComponent<Props> {
     const gaugeWidth = Math.min(dimension / 5.5, 40) / gaugeWidthReduceRatio;
     const thresholdMarkersWidth = gaugeWidth / 5;
     const text = formattedValueToString(value);
-    const fontSize = calculateFontSize(text, dimension - gaugeWidth * 3, dimension, 1, 48);
+    // This not 100% accurate as I am unsure of flot's calculations here
+    const valueWidthBase = Math.min(width, dimension * 1.3) * 0.9;
+    const fontSize = calculateFontSize(
+      text,
+      // remove gauge & marker width (on left and right side)
+      // and 10px is some padding that flot adds to the outer canvas
+      valueWidthBase - ((gaugeWidth + (showThresholdMarkers ? thresholdMarkersWidth : 0)) * 2 + 10),
```

Comment:
nit: I would have moved this to its own variable just to make it a bit more readable.